phases/register.go

```diff
@@ -1,9 +1,101 @@
 package phases
 
-import "fmt"
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
 
-// Register ...
-func Register(progress Progress) error {
-	fmt.Println("Register")
+	"github.com/bitrise-io/go-utils/log"
+)
+
+const baseURL = "https://app.bitrise.io"
+
+func startAppRegistration() (string, error) {
+	return "", nil
+}
+
+func registerSSHKey() error {
+	return nil
+}
+
+func registerWebhook(appSlug string, apiToken string) error {
+	url := fmt.Sprintf("%s/app/%s/register-webhook.json", baseURL, appSlug)
+	request, err := http.NewRequest(http.MethodPost, url, nil)
+	if err != nil {
+		return fmt.Errorf("create POST %s request: %s", url, err)
+	}
+
+	request.Header.Add("Authorization", "token "+apiToken)
+	resp, err := http.DefaultClient.Do(request)
+	if err != nil {
+		return fmt.Errorf("send POST %s request: %s", url, err)
+	}
+
+	if resp.StatusCode == 400 {
+		log.Errorf("Error registering webhook")
+		log.Warnf("Make sure you have the required access rights to the repository and that you enabled git provider integration jfor your Bitrise account!")
+		log.Warnf("Please fix your configuration and hit enter to try again!")
+
+		reader := bufio.NewReader(os.Stdin)
+
+		for {
+			_, _ = reader.ReadString('\n')
```

Comment:
Even if it's impossible to get an error for the `reader.ReadString('\n')` please still handle the error.
```
if _, err := reader.ReadString('\n'); err != nil {
 // Log
}
```